lib/vagrant-openshift/hooks/configure_aws.rb

```diff
@@ -0,0 +1,54 @@
+#--
+# Copyright 2016 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require_relative "../aws"
+
+module Vagrant
+  module Openshift
+    module Hooks
+      class ConfigureAWS
+        def initialize(app, env)
+          @app = app
+        end
+
+        def call(env)
+          if env[:machine].config.openshift.autoconfigure_aws == true
+            aws_creds = Vagrant::Openshift::Aws::aws_creds()
+            raise "AWS credentials not configured" if aws_creds.nil?
+
+            aws = env[:machine].config.vm.get_provider_config(:aws)
+            if !aws.nil?
```

Comment:
I'm not following this logic -- if we are asked to auto-configure AWS settings, why would this be `nil`, and, if it is, and we don't set up the secrets, won't everything fail when we start making AWS API calls in the future?